### PR TITLE
Enhancement-controller-mavic

### PR DESCRIPTION
**Pull request by joachimhgg**
**Description**

Example of Python controller for Mavic patrolling around the house.
This demonstrates how to go to specific world coordinates using its GPS, imu and gyroscope. The drone reaches a given altitude and patrols from waypoint to waypoint.

### CODE REVIEW

File: docs/guide/mavic-2-pro.md [hunk 1]
```diff
@@ -42,4 +42,5 @@ You will find the following sample in this folder: [WEBOTS\_HOME/projects/robots
 #### mavic\_2\_pro.wbt
 
 ![mavic_2_pro.wbt.png](images/robots/mavic-2-pro/mavic_2_pro.wbt.thumbnail.jpg) This demonstration shows a DJI Mavic 2 PRO flying over a rural area.
-After the drone has taken off, you can control it with the keyboard of your computer.
+The controller `mavic2pro.c` allows you to control the drone, after it has taken off, with the keyboard of your computer.
+The controller `mavicGotoTargets.py` shows how to move to specific world coordinates using its GPS, imu and gyroscope.
```
Comment (ad-daniel) [inline docs/guide/mavic-2-pro.md:46]: controller should probably be prefixed `mavic2pro` as the other is, not just mavic. Also the name should use snake case to be consistent with the other python controllers. As for the name itself, perhaps: `mavic2pro_waypoints` or `mavic2pro_patrol` might be better?

File: projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        # Specify the patrol coordinates
+        waypoints = [[-30, 20], [-60, 20],  [-60, 10], [-30, 5]]
+        # target altitude of the robot in meters
+        self.target_altitude = 15
+
+        while self.step(self.time_step) != -1:
+
+            # Read sensors
+            roll, pitch, yaw = self.imu.getRollPitchYaw()
+            Xpos, Ypos, altitude = self.gps.getValues()
```
Comment (ad-daniel) [inline projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py:154]: everything else is snake case, `Xpos` and `Ypos` should also be

File: projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                pow(clamped_difference_altitude, 3.0)
+
+            front_left_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input - yaw_input + pitch_input - roll_input
+            front_right_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input + yaw_input + pitch_input + roll_input
+            rear_left_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input + yaw_input - pitch_input - roll_input
+            rear_right_motor_input = self.K_VERTICAL_THRUST + \
+                vertical_input - yaw_input - pitch_input + roll_input
```
Comment (ad-daniel) [inline projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py:182]: According to the our python coding style we can have up to 128 characters per line, it shouldn't be necessary to write most of this in two lines (from L165 to L182)

File: projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        self.rear_right_motor = self.getDevice("rear right propeller")
+        self.camera_pitch_motor = self.getDevice("camera pitch")
+        self.camera_pitch_motor.setPosition(0.7)
+        motors = [self.front_left_motor, self.front_right_motor,
+                  self.rear_left_motor, self.rear_right_motor]
+        for motor in motors:
+            motor.setPosition(float('inf'))
+            motor.setVelocity(1)
+
+        self.current_pose = 6 * [0]  # X,Y,Z, yaw, pitch, roll
```
Comment (ad-daniel) [inline projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py:73]:
```suggestion
        self.current_pose = 6 * [0]  # X, Y, Z, yaw, pitch, roll
```

File: projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    def run(self):
+        t1 = self.getTime()
+
+        roll_disturbance = 0
+        pitch_disturbance = 0
+        yaw_disturbance = 0
+
+        # Specify the patrol coordinates
+        waypoints = [[-30, 20], [-60, 20],  [-60, 10], [-30, 5]]
```
Comment (ad-daniel) [inline projects/robots/dji/mavic/controllers/mavicGotoTargets/mavicGotoTargets.py:146]:
```suggestion
        waypoints = [[-30, 20], [-60, 20], [-60, 10], [-30, 5]]
```

**Review by ad-daniel — CHANGES_REQUESTED** [review]
Thank you, only a few comments

**Review by ad-daniel — DISMISSED** [review]
Thank you!

File: docs/guide/mavic-2-pro.md [hunk 6]
```diff
@@ -42,4 +42,5 @@ You will find the following sample in this folder: [WEBOTS\_HOME/projects/robots
 #### mavic\_2\_pro.wbt
 
 ![mavic_2_pro.wbt.png](images/robots/mavic-2-pro/mavic_2_pro.wbt.thumbnail.jpg) This demonstration shows a DJI Mavic 2 PRO flying over a rural area.
-After the drone has taken off, you can control it with the keyboard of your computer.
+The controller `mavic2pro.c` allows you to control the drone, after it has taken off, with the keyboard of your computer.
+The controller `mavic2pro_patrol.py` shows how to move to specific world coordinates using its GPS, imu and gyroscope.
```
Comment (omichel) [inline docs/guide/mavic-2-pro.md:46]: I believe it is better this way:
```suggestion
The `mavic2pro.c` controller allows you to control the drone, after it has taken off, with the keyboard of your computer.
The `mavic2pro_patrol.py` controller shows how to move to specific world coordinates using its GPS, imu and gyroscope.
```

**Review by ad-daniel — APPROVED** [review]
Thank you, github still having connection issues. I've been rerunning the job a few times but to no avail

**Pull request merged**